miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/UserRestControllerST.java

```diff
@@ -0,0 +1,129 @@
+import uk.ac.bbsrc.tgac.miso.webapp.controller.rest.UserRestController.PasswordChangeDto;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.*;
+import javax.ws.rs.core.MediaType;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.*;
+
+
+public class UserRestControllerST extends AbstractST {
+
+
+  private static final String CONTROLLER_BASE = "/rest/users";
+  private static final Class<UserImpl> entityClass = UserImpl.class;
+
+
+  // create, update, delete, search, change password
+  private UserDto makeCreateDto() throws Exception {
+    UserDto dto = new UserDto();
+    dto.setAdmin(false);
+    dto.setActive(true);
+    dto.setInternal(true);
+    dto.setLoginName("new");
+    dto.setFullName("new guy");
+    dto.setPassword("OntarioInstituteOfCancerResearch2025");
+    return dto;
+  }
+
+  @Test
+  @WithMockUser(username = "admin", password = "admin", roles = {"INTERNAL", "ADMIN"})
+  public void testCreate() throws Exception {
+    // must be admin to create user
+    UserImpl user = baseTestCreate(CONTROLLER_BASE, makeCreateDto(), entityClass, 200);
+    assertEquals(false, user.isAdmin());
+    assertEquals(true, user.isActive());
+    assertEquals(true, user.isInternal());
+    assertEquals("new", user.getLoginName());
+    assertEquals("new guy", user.getFullName());
+  }
+
+  @Test
+  public void testCreateFail() throws Exception {
+    // must be admin to create user
+    testCreateUnauthorized(CONTROLLER_BASE, makeCreateDto(), entityClass);
+  }
+
+
+  @Test
+  @WithMockUser(username = "admin", password = "admin", roles = {"INTERNAL", "ADMIN"})
+  public void testUpdate() throws Exception {
+    // must be admin or matching owner to update user
+
+    UserDto user = Dtos.asDto(currentSession().get(entityClass, 1));
+    user.setEmail("newemail@gmail.com");
+    UserImpl updated = baseTestUpdate(CONTROLLER_BASE, user, 1, entityClass);
+    assertEquals(user.getEmail(), updated.getEmail());
+  }
+
+  @Test
+  public void testUpdateFail() throws Exception {
+    // must be admin or matching owner to update user
+
+    UserDto user = Dtos.asDto(currentSession().get(entityClass, 1));
+    user.setEmail("newemail@gmail.com");
+    testUpdateUnauthorized(CONTROLLER_BASE, user, 1, entityClass);
+  }
+
+
+  @Test
+  @WithMockUser(username = "admin", password = "admin", roles = {"INTERNAL", "ADMIN"})
+  public void testDelete() throws Exception {
+    // must be admin to delete user
+    testBulkDelete(entityClass, 4, CONTROLLER_BASE);
+  }
+
+  @Test
+  public void testDeleteFail() throws Exception {
+    testDeleteUnauthorized(entityClass, 4, CONTROLLER_BASE);
+  }
+
+
+  @Test
+  public void testSearch() throws Exception {
+    baseSearchByTerm(CONTROLLER_BASE, "hhenderson", Arrays.asList(4));
```

Comment:
I had to open up the AbstractST class to figure out what `Arrays.asList(4))` represented. Could you make this more readily understandable by extracting that argument to a variable named something like `expectedIds` ? 

(Though it looks like that's not truly what the test is asserting; it's just confirming that the number of IDs passed in is the number of IDs returned in the search. I think it would be clearer if the argument was an integer of `expectedResultsCount` rather than a list of integers, but I'm ambivalent on whether it's worth refactoring all the tests that use this function.)